go/sqltypes/value.go

```diff
@@ -548,7 +548,12 @@ func encodeBytesSQL(val []byte, b BinWriter) {
 
 func encodeBytesSQLBytes2(val []byte, buf *bytes2.Buffer) {
 	buf.WriteByte('\'')
-	for _, ch := range val {
+	for idx, ch := range val {
+		// If \% or \_ is present, we want to keep them as is, and don't want to escape \ again
+		if ch == '\\' && idx+1 < len(val) && (val[idx+1] == '%' || val[idx+1] == '_') {
+			buf.WriteByte(ch)
+			continue
+		}
```

Comment:
I had to duplicate this code in all three encoding functions we have. I couldn't come up with a better way 😢. We can merge two of them using an interface which requires a `WriteByte` function, but the last remains separate.

Reply:
In this light, I have not merged any of these functions. Open to suggestions for improvement.

Reply:
we could extract `ch == '\\' && idx+1 < len(val) && (val[idx+1] == '%' || val[idx+1] == '_')` to a method to avoid duplicate, and send in `ch`, `idx`, `val`

Reply:
otherwise it looks good

Reply:
I tried it out, but that too doesn't seem to work all that well, since val is a byte array in two of the funcs and a string in the other, and ch is a byte in two and rune in the other. I am leaving this as is and we can iterate on this later.